src/misc/ErrorsPanel/ErrorsPanel.js

```diff
@@ -17,50 +17,65 @@ import useStyles from './style';
 
 export const ErrorsPanel = (props) => {
   const classes = useStyles();
-  const { errors, labels, onClear, buildErrorsCountLabel } = props;
+  const { errors, maxErrorsCount, labels, onClear, buildErrorsCountLabel } = props;
 
   let errorsCountLabel = labels.errorsCountLabel;
   if (buildErrorsCountLabel) {
-    errorsCountLabel = buildErrorsCountLabel(errors.length);
+    errorsCountLabel = buildErrorsCountLabel(errors.length, maxErrorsCount);
   }
 
+  let errorsBounded;
+  if (errors.length > maxErrorsCount) {
+    errorsBounded = errors.slice(0, maxErrorsCount);
+  }
```

Comment:
We could use the `splice` [method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) instead of `slice` to prevent copying the array (use splice only if the array length is greater than 100): we don't need to keep all the errors and can thus modify the initial array.
Maybe something like `errors.splice(100, Infinity)` could be used in all cases, removing the need for the `if` condition (if the array length is lower than 100, then [no elements will be deleted](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#parameters)

Reply:
After discuss, it is more simple to keep splice method.